### PR TITLE
Added abstract class and @interface snippets for new java file generation

### PR DESCRIPTION
**Pull request by ghokun**
newsnippet

This is my fix to issue #1722.

Signed-off-by: ghokun

### CODE REVIEW

File: src/fileEventHandler.ts [hunk 1]
```diff
@@ -119,9 +119,9 @@ async function handleNewJavaFiles(e: FileCreateEvent) {
             if (isModuleInfo) {
                 snippets.push(`module \${1:name} {`);
             } else if (!serverReady || await isVersionLessThan(emptyFiles[i].toString(), 14)) {
-                snippets.push(`public \${1|class,interface,enum|} ${typeName} {`);
+                snippets.push(`public \${1|class,abstract class,interface,@interface,enum|} ${typeName} {`);
             } else {
-                snippets.push(`public \${1|class ${typeName},interface ${typeName},enum ${typeName},record ${typeName}()|} {`);
+                snippets.push(`public \${1|class ${typeName},abstract class ${typeName},interface ${typeName},@interface ${typeName},enum ${typeName},record ${typeName}()||} {`);
```
Comment (testforstephen) [inline src/fileEventHandler.ts:124]: There is an extra `|` at the end, need remove it.

Comment (ghokun) [reply src/fileEventHandler.ts]: Sorry about that, fixed now.
java15

File: src/fileEventHandler.ts [hunk 2]
```diff
@@ -119,9 +119,9 @@ async function handleNewJavaFiles(e: FileCreateEvent) {
             if (isModuleInfo) {
                 snippets.push(`module \${1:name} {`);
             } else if (!serverReady || await isVersionLessThan(emptyFiles[i].toString(), 14)) {
-                snippets.push(`public \${1|class,interface,enum|} ${typeName} {`);
+                snippets.push(`public \${1|class,abstract class,interface,@interface,enum|} ${typeName} {`);
```
Comment (testforstephen) [inline src/fileEventHandler.ts:122]: i prefer to adjust the order, and put the new ones `abstract class` and `@interface` last.

**Review by testforstephen — APPROVED** [review]
LGTM

**Comment by testforstephen** [comment]
thanks @ghokun

**Comment by fbricon** [comment]
@testforstephen FYI, no need to add labels/milestones to a PR if it already has a matching issue (the issue should get those instead)

**Comment by testforstephen** [comment]
i see. thanks for the clarification.

**Pull request merged**